core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -309,8 +316,17 @@ private <O> GraphResult<O> _execute(final StoreExecuter<O> storeExecuter, final
 
         final Context clonedContext = request.getContext().shallowClone();
         final OperationChain clonedOpChain = request.getOperationChain().shallowClone();
+
+        // OpenTelemetry hooks
+        Span span = OtelUtil.startSpan(
+            this.getClass().getName(),
+            "Graph Request: " + clonedOpChain.toOverviewString());
+        span.setAttribute("gaffer.graphId", getGraphId());
+        span.setAttribute("gaffer.jobId", clonedContext.getJobId());
+        span.setAttribute("gaffer.user", clonedContext.getUser().getUserId());
+
         O result = null;
-        try {
+        try (Scope scope = span.makeCurrent()) {
```

Comment:
This `scope` object doesn't seem to be used inside the try?

Reply:
It in the try with resources block so that it is auto closed, the scope simply sets the span to be current so that it correctly makes the links between parent and child spans: https://opentelemetry.io/docs/languages/java/instrumentation/#create-spans

Reply:
I thought it would be something like that, thanks for the link. It's an interesting use for the auto close functionality, I'm not sure I like it though. Could you add a single line comment to briefly cover what's happening?

Reply:
I agree does look slightly odd but this auto links the spans, the alternative is passing and linking them manually through the Opentelementry `Context`which is lot more effort. I've added comments above relevant lines to clarify.